osu.Game/Beatmaps/Beatmap.cs

```diff
@@ -58,7 +69,27 @@ public ControlPoint TimingPointAt(double time, out ControlPoint overridePoint)
 
             return timingPoint ?? ControlPoint.Default;
         }
+    }
 
+    /// <summary>
+    /// A Beatmap containing un-converted HitObjects.
+    /// </summary>
+    public class Beatmap : Beatmap<HitObject>
+    {
+        /// <summary>
+        /// Calculates the star difficulty for this Beatmap.
+        /// </summary>
+        /// <returns>The star difficulty.</returns>
         public double CalculateStarDifficulty() => Ruleset.GetRuleset(BeatmapInfo.Mode).CreateDifficultyCalculator(this).Calculate();
+
+        /// <summary>
+        /// Converts this Beatmap to a <see cref="Beatmap{T}"/> containing another type of <see cref="HitObject"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of HitObject the new Beatmap should contain.</typeparam>
+        /// <returns></returns>
+        public Beatmap<T> ConvertTo<T>(PlayMode playMode) where T : HitObject
+        {
+            return Ruleset.GetRuleset(playMode).CreateBeatmapConverter<T>().Convert(this);

```

Comment:
Playmode is needed for obvious reasons. Generic is needed for type safety, otherwise what to return? Generic Beatmap and then cast to Beatmap<T> inside hitrenderers? The other option is to not put this in Beatmap, and just leave it in Ruleset, which might be okay since this is non-destructive anyway.